pandas/core/algorithms.py

```diff
@@ -822,9 +822,19 @@ def value_counts_arraylike(values, dropna: bool):
 
         mask = isna(values)
         if not dropna and mask.any():
+            # GH 35922. Series.sort_values is stable now, so need to
+            # append NaN counts or move to the end to make sure they are
+            # sorted toward the end when calling value_counts
             if not isna(keys).any():
-                keys = np.insert(keys, 0, np.NaN)
-                counts = np.insert(counts, 0, mask.sum())
+                keys = np.append(keys, np.NaN)
+                counts = np.append(counts, mask.sum())
+            else:
+                nan_pos = np.where(np.isnan(keys))
+                keys[nan_pos] = keys[-1]
+                keys[-1] = np.NaN
+                tmp = counts[nan_pos]
+                counts[nan_pos] = counts[-1]
+                counts[-1] = tmp
```

Comment:
All of this is to ensure that NaNs appear after other items with duplicate counts when using value_counts. Could change the tests instead. I think that it's nice to have NaNs in a familiar spot, but it's not necessary.

Reply:
Whether NaN ends up first or last or wherever among duplicates is largely irrelevant. Properly guaranteeing it being in the last position requires moving it to the end of `keys` and `counts` (recreating both arrays), and that's a performance cost with no significant upside. Better to just alter tests.